### PR TITLE
[Workspace] introducing workspace level ui settings

### PR DESCRIPTION
**Pull request by ruanyl**
When getting ui settings within a workspace, it will combine the workspace ui settings with the global ui settings and workspace ui settings have higher priority if the same setting was defined in both places

When updating ui settings within a workspace, it will update the workspace ui settings, the global ui settings will remain unchanged.

### Issues Resolved
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/6509

## Changelog
- feat: introducing workspace ui settings which allow users to customize ui settings for each individual workspace.

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/workspace/server/saved_objects/integration_tests/workspace_ui_settings_wrapper.test.ts
Comment (yujin-emma) [inline src/plugins/workspace/server/saved_objects/integration_tests/workspace_ui_settings_wrapper.test.ts:1]: could you please resolve the license header issue as it suggested

Comment (ruanyl) [reply src/plugins/workspace/server/saved_objects/integration_tests/workspace_ui_settings_wrapper.test.ts:1]: Thanks, updated ;)

**Review by BionIT — APPROVED** [review]
1 question, not a blocker

File: src/plugins/workspace/server/saved_objects/workspace_ui_settings_client_wrapper.ts [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+      options: SavedObjectsUpdateOptions = {}
+    ): Promise<SavedObjectsUpdateResponse<T>> => {
+      const { requestWorkspaceId } = getWorkspaceState(wrapperOptions.request);
+
+      /**
+       * When updating ui settings within a workspace, it will update the workspace ui settings,
+       * the global ui settings will remain unchanged.
+       */
+      if (type === 'config' && requestWorkspaceId) {
+        const configObject = await wrapperOptions.client.get<Record<string, any>>(
```
[the remark this answers is not in the document]
Comment (ruanyl) [reply src/plugins/workspace/server/saved_objects/workspace_ui_settings_client_wrapper.ts:92]: Not sure if I get the question, what are referring by `default saved object client`?

[the remark this answers is not in the document]
Comment (BionIT) [reply src/plugins/workspace/server/saved_objects/workspace_ui_settings_client_wrapper.ts:92]: Yes, want to understand what does wrapperOptions mean. I assume it is saved object client

**Review by BionIT — APPROVED** [review]

**Review by SuZhou-Joe — APPROVED** [review]

**Pull request merged**